src/Dependencies/Collections/SegmentedArray.cs

```diff
@@ -192,15 +192,20 @@ public static int BinarySearch<T>(SegmentedArray<T> array, int index, int length
 
         public static int IndexOf<T>(SegmentedArray<T> array, T value)
         {
-            return IndexOf(array, value, 0, array.Length);
+            return IndexOf(array, value, 0, array.Length, comparer: null);
         }
 
         public static int IndexOf<T>(SegmentedArray<T> array, T value, int startIndex)
         {
-            return IndexOf(array, value, startIndex, array.Length - startIndex);
+            return IndexOf(array, value, startIndex, array.Length - startIndex, comparer: null);
```

Comment:
What's the advantage of adding comparer: null? Are you just trying to avoid adding an extra stack frame while adding the capability of specifying a comparer? If so, maybe consider making comparer non-nullable and defaulting it to `EqualityComparer<T>.Default` so you can also skip to the else part of the clause.

Reply:
> maybe consider making comparer non-nullable and defaulting it to `EqualityComparer<T>.Default` so you can also skip to the else part of the clause.

This method is part of the public API surface. It's customary for public APIs to not require equality comparers be specified.